libs/BLEKit/include/AdvertisingData.h

```diff
@@ -0,0 +1,34 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <span>
+
+namespace leka {
+
+struct AdvertisingData {
+	const char *name = "Leka";	 // TODO: Get default name from configuration files
+	uint8_t battery {};
+	uint8_t is_charging {};
+	uint8_t led_red {};
+	uint8_t led_green {};
+	uint8_t led_blue {};
+
+	static inline int _advertising_data_size = 5;
+	std::array<uint8_t, 16> data {};
+
+	auto getData() -> std::span<const uint8_t>
+	{
+		auto tmp = std::to_array<uint8_t>({battery, is_charging, led_red, led_green, led_blue});
+		std::copy(tmp.begin(), tmp.end(), data.begin());
+		return std::span<uint8_t> {data.data(), std::size(tmp)};
+	}
+
+	[[nodiscard]] auto getDataSize() const -> int { return _advertising_data_size; }
```

Comment:
I was not happy with my previous solution. Your solutions in particular showed me the shortcomings of mine.
so I spent a little more time trying to find a way that would make it more future proof, avoid issues and work in our uses cases.

I've come up with the following, the only difference being the colors that you can keep the way you made them as discussed.

```cpp
 struct AdvertisingData {
    AdvertisingData(const char* n, uint8_t b, uint8_t ic, RGB c) : name(n), battery(b), is_charging(ic), color(c) {}

	const char *name {};
	uint8_t battery {};
	uint8_t is_charging {};
	RGB color {};

    auto values() {
        updateValues();
        return std::span{_values};
    }
    
    auto data() {
        updateValues();
        return std::span{_values}.data();
    }

    auto size() { return _values.size(); }

    private:
    void updateValues() {
         _values = {battery, is_charging, color.red, color.green, color.blue};
    }

    std::array<uint8_t, 5> _values = {};
};
```

the goal was:

- avoid the need to store the size
- have a well known interface with `data()` and `size()`
- have `data()` actually do what `std::array::data()` does --> pointer to first element
- have `values()` that returns a span to be able to iterate over the values
- hide the implementation details and underlying container
- be able to update one value and still get all the values with the updated one

You can see a working example with comments here: https://gcc.godbolt.org/z/WssxGv7qs